app/admin.go

```diff
@@ -112,7 +183,63 @@ func (s *Server) GetLogsSkipSend(page, perPage int) ([]string, *model.AppError)
 					if err != nil {
 						return nil, model.NewAppError("getLogs", "api.admin.file_read_error", nil, "", http.StatusInternalServerError).Wrap(err)
 					}
-					lines = append(lines, string(line))
+
+					filtered := false
```

Comment:
All this logic is deeply nested already and is adding more nesting, I think would be great to move it into helper private function (or maybe two), something like `isFilteredByLogLevel` and `isFilteteredByDate`, that would give you the same thing, but a more clear code.

Reply:
So I broke everything into helper functions, definitely looks cleaner. The one thing I didn't do was check first if we should filter or not. The reason was just to separate concerns. I don't think it'll be a huge performance hit to avoid this, but we can revisit if we think it's going to be an issue to not check first.